spec/apis/path-to-active-when.spec.js

```diff
@@ -2,182 +2,121 @@ import { pathToActiveWhen } from "single-spa";
+      "http://app.com/pathname?query-string=1": false,
+    });
+
+    expectPathToMatch("/pathname/:dynamic/", {
+      "http://app.com/pathname/123/": true,
+      "http://app.com/pathname/123/?query-string=1": true,
+      "http://app.com/pathname/123/extra": true,
+      "http://app.com/pathname/123/extra?query-string=1": true,
+      "http://app.com/pathname/123": false,
+      "http://app.com/pathname/123?query-string=1": false,
+    });
+
+    expectPathToMatch("/#/pathname", {
+      "http://app.com/#/pathname": true,
+      "http://app.com/#/pathname?query-string=1": true,
+      "http://app.com/#/pathname/": true,
+      "http://app.com/#/pathname/?query-string=1": true,
+      "http://app.com/#/pathname/anything/everything": true,
+      "http://app.com/#/pathname/anything/everything?query-string=1": true,
+      "http://app.com?query-string=1/#/pathname/anything/everything?query-string=1": true,
+      "http://app.com?query-string=1#/pathname/anything/everything?query-string=1": true,
+    });
+
+    expectPathToMatch("/#/pathname/:dynamic/notDynamic", {
+      "http://app.com/#/pathname/1/notDynamic": true,
+      "http://app.com/#/pathname/1/notDynamic/": true,
+      "http://app.com/#/pathname/1/notDynamicExtra": false,
+      "http://app.com/#/pathname//notDynamic/anything/everything": false,
+    });
+
+    expectPathToMatch("/pathname/:dynamic/notDynamic", {
+      "http://app.com/pathname/1/notDynamic": true,
+      "http://app.com/pathname/1/notDynamic/anything/everything": true,
+      "http://app.com/pathname//notDynamic": false,
+    });
+
+    expectPathToMatch("", {
+      "http://app.com": true,
+      "http://app.com/": true,
+      "http://app.com/anything/everything": true,
+    });
+
+    expectPathToMatch("/", {
+      "http://app.com/": true,
+      "http://app.com/anything/everything": true,
+    });
+
+    expectPathToMatch("/:dynamic/:dynamic", {
+      "http://app.com/1/1": true,
+      "http://app.com/1/1/": true,
+      "http://app.com/1//": false,
+      "http://app.com/1/": false,
+      "http://app.com/1": false,
+    });
+
+    expectPathToMatch("/:dynamic/:dynamic/anything/everything", {
+      "http://app.com/1/1/anything/everything": true,
+    });
+
+    expectPathToMatch("/pathname#/subpath/:dynamic", {
+      "http://app.com/pathname#/subpath/1/with/other/things": true,
+      "http://app.com/#/subpath/1/with/other/things": false,
+    });
+
+    expectPathToMatch("/#/subpath/:dynamic", {
+      "http://app.com/#/subpath/1/with/other/things": true,
+      "http://app.com/#/subpath/1": true,
+      "http://app.com/subpath/1/with/other/things": false,
+    });
+
+    // Impossible paths due to URL always goin from root 'app.com' -> 'app.com/'
+    expectPathToMatch("#/subpath/:dynamic", {
+      "http://app.com#/subpath/1/with/other/things": false,
+      "http://app.com#/subpath/1": false,
+    });
+
+    expectPathToMatch("pathname", {
+      "http://app.com/pathname": false,
```

Comment:
Why is this false? Because it's missing the leading slash?

Reply:
Yes, when the appToActiveWhen manipulates the `location.href`, there will always give a route starting with `/`, so unless we improve these cases where the path doesn't start with '/', in more real life scenarios it wouldn't pass